packages/block-library/src/search/index.php

```diff
@@ -190,24 +199,66 @@ function styles_for_block_core_search( $attributes ) {
 	if ( $has_border_radius ) {
 		// Shared style for button and input radius values.
 		$border_radius   = $attributes['style']['border']['radius'];
-		$shared_styles[] = sprintf( 'border-radius: %spx;', esc_attr( $border_radius ) );
```

Comment:
I split the shared style up because the style are no longer shared when the button is on the inside.